From f82d70a644b7695142b882fbb40d8f8cc70d38db Mon Sep 17 00:00:00 2001
From: Kamil Dziezyk <kamil.dziezyk@arm.com>
Date: Wed, 25 May 2022 19:22:14 +0100
Subject: [PATCH] features/xen: Move x86 configs to separate file

These configs are:
 - HYPERVISOR_GUEST
 - PCI_XEN
 - XEN_ACPI_PROCESSOR
 - XEN_DEBUG_FS
 - XEN_MCE_LOG
 - XEN_PCIDEV_BACKEND
 - XEN_PCIDEV_FRONTEND
 - XEN_PVHVM
 - XEN_SAVE_RESTORE

This patch also replaces config 'XEN_SCRUB_PAGES' with the
'XEN_SCRUB_PAGES_DEFAULT', because it was renamed since 4.19 release.

Signed-off-by: Kamil Dziezyk <kamil.dziezyk@arm.com>
---
 features/xen/xen-x86.cfg | 11 +++++++++++
 features/xen/xen.cfg     | 11 +----------
 features/xen/xen.scc     |  4 ++++
 3 files changed, 16 insertions(+), 10 deletions(-)
 create mode 100644 features/xen/xen-x86.cfg

diff --git a/features/xen/xen-x86.cfg b/features/xen/xen-x86.cfg
new file mode 100644
index 00000000..39e6993f
--- /dev/null
+++ b/features/xen/xen-x86.cfg
@@ -0,0 +1,11 @@
+# SPDX-License-Identifier: MIT
+
+CONFIG_HYPERVISOR_GUEST=y
+CONFIG_PCI_XEN=y
+CONFIG_XEN_ACPI_PROCESSOR=m
+# CONFIG_XEN_DEBUG_FS is not set
+CONFIG_XEN_MCE_LOG=y
+CONFIG_XEN_PCIDEV_BACKEND=m
+CONFIG_XEN_PCIDEV_FRONTEND=y
+CONFIG_XEN_PVHVM=y
+CONFIG_XEN_SAVE_RESTORE=y
diff --git a/features/xen/xen.cfg b/features/xen/xen.cfg
index 1f414c08..7da87825 100644
--- a/features/xen/xen.cfg
+++ b/features/xen/xen.cfg
@@ -1,14 +1,8 @@
-CONFIG_HYPERVISOR_GUEST=y
 CONFIG_PARAVIRT=y
 CONFIG_XEN=y
 CONFIG_XEN_DOM0=y
-CONFIG_XEN_PVHVM=y
-CONFIG_XEN_SAVE_RESTORE=y
-# CONFIG_XEN_DEBUG_FS is not set
 CONFIG_MMU_NOTIFIER=y
 CONFIG_HIBERNATE_CALLBACKS=y
-CONFIG_PCI_XEN=y
-CONFIG_XEN_PCIDEV_FRONTEND=y
 CONFIG_SYS_HYPERVISOR=y
 CONFIG_XEN_BLKDEV_FRONTEND=y
 CONFIG_XEN_BLKDEV_BACKEND=m
@@ -28,7 +22,7 @@ CONFIG_FB_SYS_FOPS=y
 CONFIG_FB_DEFERRED_IO=y
 CONFIG_XEN_FBDEV_FRONTEND=y
 CONFIG_XEN_BALLOON=y
-CONFIG_XEN_SCRUB_PAGES=y
+CONFIG_XEN_SCRUB_PAGES_DEFAULT=y
 CONFIG_XEN_DEV_EVTCHN=y
 CONFIG_XEN_BACKEND=y
 CONFIG_XENFS=y
@@ -38,10 +32,7 @@ CONFIG_XEN_XENBUS_FRONTEND=y
 CONFIG_XEN_GNTDEV=m
 CONFIG_XEN_GRANT_DEV_ALLOC=m
 CONFIG_SWIOTLB_XEN=y
-CONFIG_XEN_PCIDEV_BACKEND=m
 CONFIG_XEN_PRIVCMD=y
-CONFIG_XEN_ACPI_PROCESSOR=m
-CONFIG_XEN_MCE_LOG=y
 CONFIG_XEN_HAVE_PVMMU=y
 CONFIG_XEN_PVCALLS_FRONTEND=y
 CONFIG_XEN_PVCALLS_BACKEND=y
diff --git a/features/xen/xen.scc b/features/xen/xen.scc
index b588e5df..e7c85a1b 100644
--- a/features/xen/xen.scc
+++ b/features/xen/xen.scc
@@ -2,3 +2,7 @@ define KFEATURE_DESCRIPTION "Xen Kernel Support"
 define KFEATURE_COMPATIBILITY arch
 
 kconf non-hardware xen.cfg
+
+if [ "$KARCH" = "x86_64" ] || [ "$KARCH" = "i386" ]; then
+    kconf non-hardware xen-x86.cfg
+fi
-- 
2.17.1

